### PR TITLE
[CDF-23592] 😋 Move lookup to Toolkit Client

### PR DESCRIPTION
**Pull request by doctrino**
# Description

Please describe the change you have made.

## Checklist

- [ ] Tests added/updated.
- [ ] Run Demo Job Locally.
- [ ] Documentation updated.
- [ ] Changelogs updated in [CHANGELOG.cdf-tk.md](https://github.com/cognitedata/toolkit/blob/main/CHANGELOG.cdf-tk.md).
- [ ] Template changelogs updated in [CHANGELOG.templates.md](https://github.com/cognitedata/toolkit/blob/main/CHANGELOG.templates.md).
- [ ] Version bumped.
  [_version.py](https://github.com/cognitedata/toolkit/blob/main/cognite/cognite_toolkit/_version.py) and
  [pyproject.toml](https://github.com/cognitedata/toolkit/blob/main/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite_toolkit/_cdf_tk/loaders/_base_loaders.py [hunk 1]
```diff
@@ -267,20 +267,16 @@ def _split_ids(cls, ids: T_ID | int | SequenceNotStr[T_ID | int] | None) -> tupl
         raise ValueError(f"Invalid ids: {ids}")
 
     def load_resource_file(
-        self, filepath: Path, ToolGlobals: CDFToolConfig, skip_validation: bool
+        self, filepath: Path, ToolGlobals: CDFToolConfig, is_dry_run: bool = False
```
Comment (doctrino) [inline cognite_toolkit/_cdf_tk/loaders/_base_loaders.py:270]: Renamed this one to make it more aligned with the CLI such that it is easier to remember what it is doing.

File: tests/test_unit/test_cli/test_build_deploy_snapshots/cdf_common.yaml [hunk 2]
```diff
@@ -10,7 +10,7 @@ Database:
 - name: ingestion
 ExtractionPipeline:
 - createdBy: unknown
-  dataSetId: 0
+  dataSetId: 316416785021717804
```
Comment (doctrino) [inline tests/test_unit/test_cli/test_build_deploy_snapshots/cdf_common.yaml:13]: The new mocking hashes the `dataSetExternalId` string and produces a consistent 16 digit int number.

File: cognite_toolkit/_cdf_tk/client/api_client.py [hunk 3]
```diff
@@ -1,15 +1,13 @@
...
 
-from cognite.client import ClientConfig, CogniteClient
+from cognite.client import ClientConfig
+from cognite.client._api_client import APIClient
 
-from .api.dml import DMLAPI
-from .api.location_filters import LocationFiltersAPI
-from .api.robotics import RoboticsAPI
+if TYPE_CHECKING:
+    from ._toolkit_client import ToolkitClient
```
Comment (ronpal) [inline cognite_toolkit/_cdf_tk/client/api_client.py:7]: What's the purpose of this construct? Under what condition can this be false?

**Review by ronpal — APPROVED** [review]

File: cognite_toolkit/_cdf_tk/client/api_client.py (hunk 3 above)
Comment (doctrino) [reply cognite_toolkit/_cdf_tk/client/api_client.py:7]: Avoid circular imports. It is only used by static type checkers such as mypy

**Pull request merged**